Fleece/Support/DateFormat.hh

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include "fleece/slice.hh"
+#include <chrono>
+#include <optional>
+
+namespace fleece {
+
+    static constexpr int64_t kInvalidDate = INT64_MIN;
+
+    /** Maximum length of a formatted ISO-8601 date. (Actually it's a bit bigger.) */
+    static constexpr size_t kFormattedISO8601DateMaxSize = 40;
+
+    class DateFormat {
+      public:
+        static std::optional<DateFormat> parse(slice formatString);
+
+        /** Formats a timestamp (milliseconds since 1/1/1970) as an ISO-8601 date-time.
+        @param buf  The location to write the formatted C string. At least
+                    kFormattedISO8601DateMaxSize bytes must be available.
+        @param timestamp  The timestamp (milliseconds since 1/1/1970).
+        @param asUTC  True to format as UTC, false to use the local time-zone.
+        @param fmt The model to use for formatting (i.e. which portions to include).
+                      If null, then the full ISO-8601 format is used
+        @return  The formatted string (points to `buf`). */
+        static slice format(char buf[], int64_t timestamp, bool asUTC, std::optional<DateFormat> fmt);
+
+        /** Formats a timestamp (milliseconds since 1/1/1970) as an ISO-8601 date-time.
+        @param buf  The location to write the formatted C string. At least
+                    kFormattedISO8601DateMaxSize bytes must be available.
+        @param timestamp  The timestamp (milliseconds since 1/1/1970).
+        @param tzoffset   The timezone offset from UTC in minutes
+        @param fmt The model to use for formatting (i.e. which portions to include).
+                      If null, then the full ISO-8601 format is used
+        @return  The formatted string (points to `buf`). */
+        static slice format(char buf[], int64_t timestamp, std::chrono::minutes tzoffset,
+                            std::optional<DateFormat> fmt);
+
+        static DateFormat kISO8601;
+
+        enum class Year : uint8_t {};
+        enum class Month : uint8_t {};
+        enum class Day : uint8_t {};
+        enum class Hours : uint8_t {};
+        enum class Minutes : uint8_t {};
+        enum class Seconds : uint8_t {};
+        enum class Millis : uint8_t {};
+        enum class Timezone : uint8_t { NoColon, Colon };
+        enum class Separator : char { Space = ' ', T = 'T' };
+
+        struct YMD {
```

Comment:
It looks as though the only parts of YMD and HMS that are used by this PR are the separators. I understand wanting room to grow, but all the work of defining types and fields and structs that aren't used yet seems too much ... It seems like each of these structs could be reduced to the Separator enum for now, and it wouldn't be difficult to refactor them back to structs in the future if/when we need it.